core/src/test/java/io/grpc/internal/ClientCallImplTest.java

```diff
@@ -0,0 +1,104 @@
+package io.grpc.internal;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Matchers.isA;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.util.concurrent.MoreExecutors;
+
+import io.grpc.CallOptions;
+import io.grpc.ClientCall;
+import io.grpc.Codec;
+import io.grpc.DecompressorRegistry;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.grpc.MethodDescriptor.Marshaller;
+import io.grpc.MethodDescriptor.MethodType;
+import io.grpc.internal.ClientCallImpl.ClientTransportProvider;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.ArgumentCaptor;
+
+import java.io.InputStream;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * Test for {@link ClientCallImpl}.
+ */
+@RunWith(JUnit4.class)
+public class ClientCallImplTest {
+
+  private final SerializingExecutor executor =
+      new SerializingExecutor(MoreExecutors.newDirectExecutorService());
+  private final ScheduledExecutorService deadlineCancellationExecutor =
+      Executors.newScheduledThreadPool(0);
+
+  @Before
+  public void setUp() {
+    DecompressorRegistry.register(new Codec.Gzip(), true);
+  }
+
+  @Test
+  public void advertisedEncodingsAreSent() {
+    MethodDescriptor<Void, Void> descriptor = MethodDescriptor.create(
+        MethodType.UNARY,
+        "service/method",
+        new TestMarshaller<Void>(),
+        new TestMarshaller<Void>());
+    final ClientTransport transport = mock(ClientTransport.class);
+    ClientTransportProvider provider = new TestClientTransportProvider() {
```

Comment:
nit: you're overriding the only method ... maybe just `new ClientTransportProvider`?